src/source/Sdp/Deserialize.c

```diff
@@ -74,74 +99,76 @@ STATUS deserializeSessionDescription(PSessionDescription pSessionDescription, PC
 {
     ENTERS();
     STATUS retStatus = STATUS_SUCCESS;
-    PCHAR curr, tail, next;
-    UINT32 lineLen;
-    CHK(sdpBytes != NULL, STATUS_SESSION_DESCRIPTION_INVALID_SESSION_DESCRIPTION);
+    SdpResult_t sdpResult = SDP_RESULT_OK;
+    SdpDeserializerContext_t ctx;
+    CHAR* pValue;
+    SIZE_T valueLength;
+    UINT8 type;
+    SIZE_T minLength;
 
-    curr = sdpBytes;
-    tail = sdpBytes + STRLEN(sdpBytes);
+    CHK(sdpBytes != NULL, STATUS_SESSION_DESCRIPTION_INVALID_SESSION_DESCRIPTION);
 
-    while ((next = STRNCHR(curr, tail - curr, '\n')) != NULL) {
-        lineLen = (UINT32) (next - curr);
+    sdpResult = SdpDeserializer_Init(&ctx, sdpBytes, STRLEN(sdpBytes));
+    CHK(sdpResult == SDP_RESULT_OK, convertSdpErrorCode(sdpResult));
 
-        if (lineLen > 0 && curr[lineLen - 1] == '\r') {
-            lineLen--;
-        }
+    for (; sdpResult == SDP_RESULT_OK;) {
```

Comment:
seems odd to use `for` loop here instead of `while`?

Reply:
You're right! Great catch! Updated to while loop.